### PR TITLE
Licence management with mycila

### PR DESCRIPTION
**Pull request by jhertout**
Configure license management using [https://mycila.carbou.me/license-maven-plugin/#detailed-maven-documentation](https://mycila.carbou.me/license-maven-plugin/#detailed-maven-documentation) 

Only configure generation for our code files. Do not modify CodeNarc source files.

### CODE REVIEW

**Comment by dedece35** [comment]
Hi @jhertout,
there is a lot of modifications in this PR :(
are only licence headers modified ? if yes ... ok for me.

but the github action is in error :(

**Comment by jhertout** [comment]
Hi @dedece35,

Yes, the true file to review is only the pom file. The other files have only their license header modified the same way thanks to "mycila".

For the Github action, I see that the build failed. It does not seem related to this PR but I will look to fix it.

**Review by dedece35 — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -162,6 +162,42 @@
...
+				<plugin>
+					<groupId>com.mycila</groupId>
+					<artifactId>license-maven-plugin</artifactId>
+					<version>4.1</version>
+					<configuration>
+						<properties>
+							<project.name>ecoCode Android plugin</project.name>
+							<project.description>Provides rules to reduce the environmental footprint of your Android applications</project.description>
+							<email>contact@ecocode.io</email>
+							<year>2020</year>
```
Comment (utarwyn) [inline pom.xml:175]: Why do you want to set the license year to 2020 and not leave it as the default (the current year)?

Comment (jhertout) [reply pom.xml:175]: I never know if it is the creation year or the last publication year that must be set in the copyright. I do not know how it is important so I prefered to set the the creation year. However if you are sure that it is the current year that we must set, I can change the date.

**Pull request merged**